### ISSUE TITLE
Ring is displayed even for contact with an ENS name set

### ISSUE THREAD

**Issue by ilmotta** [status-im/status-react#20617]
# Bug Report

## Problem 

Rings should not be displayed on contacts with an ENS name set and also not shown for the currently logged account. The expected behavior is a feature to help users quickly differentiate contacts with ENS names because these names are more trustworthy than arbitrary alias/display names set by users.

Bug first reported in this issue https://github.com/status-im/status-mobile/issues/20511.

### Expected behavior

The expected behavior is the same as in status-desktop, i.e. rings are not displayed on contacts with ENS name set and are not displayed in screens showing the currently logged account avatar.

### Actual behavior

There are multiple places to check/fix for contacts with ENS names (possibly more):

- Member list in channels.
- Profile screen avatar.
- Edit Profile screen.
- Home screen profile avatar.

### Solution

Consider not duplicating conditionals/checks to decide if the ring should be displayed. Consider making this logic implicit for every avatar displayed because this business rule has no exception.

### Reproduction

1. Use an account with ENS name (has to be done in status-desktop and synced on mobile).
2. Join a community.
3. Check the avatar in a channel member list contains a ring.

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

**Comment by churik** [comment]
postponed due to limited capacity for 2.30

**Comment by ilmotta** [comment]
@status-im/design-team @jrainville @churik @shivekkhurana @iurimatias @flexsurfer @sunleos

I'm taking the opportunity in this issue to ask you all, what's the real value provided by having rings? Do you think the ring solves the user's problem effectively of discerning contacts? Here's a constructive feedback from one user https://discord.com/channels/1210237582470807632/1210237584287076355/1277973530918523011 `theres been a lot of fakes and no way to know who is real and who isn't. The ring of colors around a users profile is completely useless because no one is going to memorize those`

I'm in favor of completely removing it from the Status apps to simplify the experience, speed up rendering and eliminate sources of bugs.

There are a few usability issues from the top of my head, but maybe there's more:

- When listing contacts, the rings are too thin and discerning colors even in a large iPhone is somewhat impractical.
- Contacts with ENS name are already [shown with a green check](https://github.com/user-attachments/assets/10986e76-6276-437d-963d-74f79dd190a5). This is simple and effective.
- When listing contacts in mentions or in the members' list, the contact status indicator stays on top of the ring color, thus nullifying its value because the user can't see all colors.
- Colors are unreliable (e.g. we know how difficult it is to print accurate colors). Another example, many people (especially men) have some degree of color blindness. Some of these users would make the wrong decision about contacts based on the colors.

Even if this is not the best place for a discussion or to make a decision, I'm raising it here first for transparency and durability. We can of course continue the discussion in the chat where we can ping other folks like Volodymyr.

**Comment by flexsurfer** [comment]
my opinion remains unchanged from the beginning, rings are useless and only create unnecessary noise

**Comment by benjthayer** [comment]
I personally don't find too much value in them atm. I think having a robust way for users to self-mark known contacts as verified is more valuable and extensible. We also have the emoji hash and chat keys (the snippet for which present next to each message) which are unique identifiers to help weed out impersonators. Users can easily share the chatkey of the real member so they can differentiate an impersonator. 

The colour blindness factor and the fact that the online indicator somewhat obscures the ring are also good points @ilmotta 

I wasn't here when they were introduced however so I'm unsure if there is some rationale we're missing?

**Comment by ilmotta** [comment]
> I think having a robust way for users to self-mark known contacts as verified is more valuable and extensible.

💯 @benjthayer 

------

I found the original issue which implemented first in status-mobile https://github.com/status-im/status-mobile/issues/13334. Over the status-desktop repo I found this issue created a few months before the mobile one https://github.com/status-im/status-desktop/issues/4782.

The rationale is clear to me, but the effectiveness of this feature is not. Volo also not keen on this feature.

> As a user
> I want other users to display their identity ring in chat
> So that I can make sure they are not impersonating someone else
>
> -- https://github.com/status-im/status-mobile/issues/13334

@iurimatias and @osmaczko were involved as well in the old desktop issue https://github.com/status-im/status-desktop/issues/4782. Do you have context to help us decide what to do with this feature?

**Comment by jrainville** [comment]
I was in the design calls whens the rings were discussed and when it was first implemented.

One reason for the existence of the ring was that the emoji hash only covers half the bytes of the pubkey. The Ring covers the other half.
What that means is that if we only have the emoji hash, there is a chance that two people will have the same hash.

In practice, this is very negligible. It's not like people can choose their pubkey/hash, so someone would need to be **very** lucky to find the same hash as someone to impersonate.

As for the ring itself, I have grown fond of it, it looks kinda cool IMO. It fits a bit the new colourful design we have and since I have an ENS name now, I find my icon to be a bit "empty" now.
This is just subjective thoughts and can be resolved with just new designs, like adding a ring to all icons, just not a "special" ring.

As for trying to foil impersonation, I did use the ring once,  but it was super shallow. I checked the top and bottom colours only. It's impossible to remember all the colors.
In the end, the pubkey is the best and safest way to compare two users.

In conclusion, while I kinda like the ring right now, I do see that it doesn't serve any purpose now.

IMO, since we already have it, we could keep it, but remove all the logic of removing it when someone has an ENS, since it's not intuitive at all and it just looks weird.
We could stop advertising the ring as an impersonation barrier, but instead just say it's a personalized ring, where everyone has a unique one. It feels a bit like a rare NFT that we all have 😄 

If we decide to completely remove it to get rid of tech debt, I won't be mad either.

**Comment by benjthayer** [comment]
Great summary @jrainville thanks! Good to get that context. 

To add another purely aesthetic point to this conversation - when we eventually implement collectible avatars, and even before, its possible/likely that people will be frustrated about how the identicon ring may clash with their chosen avatar/NFT. I have an expensive NFT I want to flex as my profile picture but I'm unlucky enough to have been given a mainly brown and yellow identicon ring that clashes badly with the colours in the NFT for example.

To your point @jrainville if the identicon ring only covers half of the pubkey and the emoji hash the other, in order to truly be able to identify a user as legit, I would have to check both of these visual artefacts together. Not only that, but I would need some trusted party or source to check these assets against - I can't hold these 2 visual artefacts in memory, so I would need a screenshot of the legitimate artefacts to verify against. This feels like it might be perhaps an impractical overhead for the user which in reality will only be used a handful of times if at all? Is the chatkey just as sufficient in identifying the user?

**Comment by jrainville** [comment]
> Is the chatkey just as sufficient in identifying the user?

Yes, the chatkey is 100%* unique. If someone else has the same chatkey, they have hacked your account 😂 

* the pubkey is. The chatkey is a compressed version of the pubkey, so I'm not sure if it's 100%, but close enough

**Comment by sunleos** [comment]
Hey folks,

It is Volo :) 

Just read all the comments above. It is a great discussion 👍 
While I agree it is kind of cool and the intention behind implementing it was really great, but I completely resonate with the comments above that this feature doesn't prove to bring much user value. I feel it is reasonable to remove it (if it is not a lot of work) so we can simplify our solution a bit and focus more on other features that could bring more value to our users, but feel free to disagree here. Appreciate your comments/thoughts.

**Comment by iurimatias** [comment]
The reason fro the ring (and emoji hash) was due to the introduction of usernames, at the time this was quite a big deal and a change that introduced security concerns due to how easy one could impersonate anyone.

If we remove the ring a change we could do to somewhat compensate is replace the first letters in the preview with the actual key part that is different (note the `zQ3` which is always the same for any accounts)

![Image](https://github.com/user-attachments/assets/3c23eb22-641f-44fb-b00f-45e8cd4764d5)

**Comment by iurimatias** [comment]
>  Is the chatkey just as sufficient in identifying the user?

the FULL chatkey yes, but the preview key as it currently shows in the chat messages does not have enough entropy to be really secure (specially if 3 letters going to waste currently with that zQ3)

**Comment by iurimatias** [comment]
to further elaborate the issue with the lack of emoji hash, ring, and key preview is tied to insufficient entropy.

We used to have three-word usernames a bit like bip39 which didn't have this issue. We then introduced custom userrnames and there were concerns about impersonation. The security team strongly objected at the time. To strike a balance, we implemented measures like the emoji hash, ring, and key preview, which, when combined, provide enough entropy to make attacks difficult, though still not as secure as revealing the full key. For example, if only one letter of the key was shown, an attacker could simply keep generating accounts until they matched that letter. As more letters or elements are added, the difficulty increases, and with all three elements (emoji hash, ring, and key preview) together, impersonation becomes practically infeasible.

I'm not against removing the ring per se, but we need to take take this into account and perhaps come up with a solution to compensate. One of the suggestions is to change the key to remove `zQ3` but even this is somewhat insufficient.

**Comment by ilmotta** [comment]
Thank you all for sharing so many interesting opinions and a history lesson around the problem!

I think we can all agree the rings feature, to put it lightly, is a _questionable_ solution 🪰 to an important problem. There are other UX issues we haven't discussed, such as, very low contrast for some ring colors on top of a dark background.

Instead of trying to justify the existence of the rings, to me, a good outcome from this discussion is that we should at least descope/deprecate the rings and consider new solutions when priorities allow. I'd remove the feature entirely, but I think we can leave this decision for another moment. If descoping is unreasonable to you, please do share. But descoping now means this `status-mobile` bug #20617 won't be fixed and we can park it. Already a win (less wasted effort).

Solutions that rely on the user to constantly compare against a known good list (that they manually manage) of trusted keys/rings/etc doesn't work well from my experience. I really can't see how any user can sustainably do this. For instance, if a user in the Status community uses my current alias `icaro`, I bet almost nobody will realize, with or without the rings, that I'm being impersonated. Probably nobody will have stored my correct key/emojis/ring to be able to compare. We can sometimes use the chat history with the person to identify a fake, but this is not always viable and it's cumbersome. The best I did so far was to use `https://contacts.status.im` to compare keys when I need to be sure, but this is very time consuming.

The other alternative I use is to add a nickname following my own standard prefix in contacts I have verified or talk the most. This is the closest I got to your idea @benjthayer of `I think having a robust way for users to self-mark known contacts as verified is more valuable and extensible`, but still far from ideal.

-----

Overall, the current situation in the mobile app is not great to protect users from impersonation. When I open any contact, all I can see is the emoji hash and the ring ([screenshot](https://github.com/user-attachments/assets/479fb6e1-ef58-47ee-8f10-10fc220b6242)). This could be seen as a bug because the user can't even see the compressed key in its short form. The compressed key is only accessible if they 1) open the contact/profile drawer 2) press `Show QR code` 3) copy the link to the profile 4) extract from the URL the compressed key (after the `#`).

**Comment by xAlisher** [comment]
I believe the rings no longer provide much value, especially given better alternatives like the ENS and unique chat keys, which serve as more reliable identifiers:

- **Usability**: Rings are visually impractical, especially on smaller screens, and can be confusing for users, particularly those with color blindness.
- **Security**: While the rings were originally introduced as a security feature, they haven't significantly improved security in practice.

### Alternatives:
- **Self-marking contacts**: This isn't effective in crypto communities, where users are often strangers.
- **Contact list**: Useful for one-on-one chats but not suitable for communities.
- **ID verification**: Removed as it was too complex for the perceived value and doesn’t solve the problem in communities.

The main issue remains: identifying imposters of admins or active users in communities. Impersonation of admins can be addressed by highlighting roles with a color or icon (similar to Discord), but this icon needs to be visible in chats, not just the member list.

When it comes to community members, spotting imposters relies on users' memory of conversations rather than technological solutions. Imposters can easily copy names and profile pictures, but the real proof of authenticity lies in message history.

From my experience moderating large Discord servers, I often encountered active users being impersonated. I used the following tools:

1. **Notes**: In Discord, I added notes to user profiles proactively, though this isn't always feasible.
2. **Message history**: The most effective method I used was checking message history. By reviewing the user’s past posts, I could quickly identify whether they were legitimate or not.

### My suggestion:
1. **Display community join date**: In the hover menu, alongside the display name and chat key, show the date the user joined the community. While this won’t prevent long-term imposters who change names later, it will catch most cases.
  
2. **Quick message history access**: Provide an option to glance at a suspicious user’s message history. Perhaps show some stats like the number of messages or a link to search all their messages, or at least their last N messages.

Since rings aren’t effective, and no one remembers chat keys, I believe this approach, grounded in users' own observations and responsibility rather than technology alone, would be more practical and aligned with our values.

@ilmotta @iurimatias @churik @VolodLytvynenko

**Comment by churik** [comment]
Thank you for discussion!
My 5 cents:

> I think having a robust way for users to self-mark known contacts as verified is more valuable and extensible. 

I believe we should add the "Mark as trusted" feature like desktop has  (https://github.com/status-im/status-desktop/issues/16392 + https://github.com/status-im/status-mobile/issues/21393) 

I largely agree that the rings feature is not providing the intended value and is cluttering the user interface. Therefore, we should consider the challenges associated with removing it from all platforms and adjusting both applications, including onboarding, chat views, profiles, etc. Additionally, we would need to modify the chat key display to facilitate easier comparison in the UI. However, this assessment is largely speculative, as we have not conducted a focus group to evaluate this specific feature. 
My main concern is that its **removal may also require a significant amount of work.**

as for @xAlisher comment:
> 1) Display community join date

Might be resolved when we align mobile / desktop apps with profile showcase (unfortunately I don't know ETA) as desktop already has a feature where user can show their communities. 
As for communities admins, I'd suggest to display them as an **admins** (it is just not done on mobile) as it is already done on desktop.

> 2 Quick message history access: 

Frankly from my POV just adds more complexity, as I believe that showcase + marking as trusted + show admins will be enough + any unique identifier (ring / chat key) will be enough.

@ilmotta I can create an issues to align these with the desktop app if it makes sense.

**Comment by jrainville** [comment]
> I believe we should add the "Mark as trusted" feature like desktop 

We currently only have "mark as **un**trusted". It might be valuable to have "mark as trusted" too. The backend logic is already all there since we used to have ID verification that did that. we just don't need to have all the complex flow of asking questions.

> Message history: The most effective method I used was checking message history. By reviewing the user’s past posts, I could quickly identify whether they were legitimate or not.

We have this feature currently, but only for admins and you need to access it from the Admin panel.

So if we wanted, at least on Desktop, it would be relatively easy to add this for all users.

> Display community join date

This could be simple to do as well. It needs changes in the control node code, but nothing big.

### Other options

Some options that were brought by @iurimatias or someone else when we discussed this in our meeting:

- show an icon or something when someone has a display name that is identical or similar to one of your contacts
- show an icon that shows if someone is a new member. It's similar to the join date, but extends it to be both a "say welcome to this person", but also a "watch out this person just joined"

Obviously, in both cases, if you mark them as trusted, the icons disappear.

### Conclusion

While finding better ways to catch impersonators is a good idea, removing the ring and implementing this is probably not the top priority. We still have a couple pressing bugs and performance issues we need to focus on.

Though I agree with @ilmotta that this discussion means that it's useless to fix issues to the identity ring for now, since we seem to all be in agreement that it doesn't fulfill its purpose well.

I guess this issue could be renamed to "Ideas to stop Impersonation" 😄

**Comment by osmaczko** [comment]
Thank you for the thoughtful discussion. However, I feel we may have deviated from the main point. While we all agree that the color ring, as currently implemented with numerous segments and colors, is ineffective—no one can subconsciously remember it well enough to detect impersonators—this is not the core issue at hand. The real question is whether we should remove the ring entirely or address the conditions under which it should appear on mobile.

I don't have a strong opinion, and I may be slightly biased since I implemented both the color ring and the emoji hash. That said, I believe the decision comes down to weighing the potential benefits and drawbacks.

#### Benefits of removing the feature:
- Reduced resource usage on mobile (though how much effort would fixing the issue really require?).
- Decreased overall complexity (though this might eventually need to be reintroduced with another anti-impersonation mechanism).
- Lower maintenance burden on both desktop and mobile (although based on the status-desktop experience, I would estimate this burden to be minimal).

#### Drawbacks of removing the feature:
- Some resources will still be consumed when removing the rings and re-testing the apps, both on mobile and desktop.
- Loss of the unique avatar look that fits our colorful design(?). Some users might already be used to it, so maybe we should ask for their feedback before removing it.
- We lose the illusion of having an anti-impersonation feature, and in some cases, an illusion might be better than nothing.


Same as @jrainville, I suggest opening a separate discussion to address impersonation mitigation in Status more broadly. While there are some great ideas here, I don't believe this should be our main focus in the next few months, as we have more pressing priorities on our roadmap.

**Comment by ilmotta** [comment]
The discussion started with the comment https://github.com/status-im/status-mobile/issues/20617#issuecomment-2392604883 asking what's the value provided to users. The short answer as far as I understood is that the rings provide little to no value to solve impersonation and many different solutions are possible. Based on at least one user feedback, the feature provides negative value because it's confusing. Clearly, the solution with rings is not human-friendly, although visually appealing.

#### Effort to disable rings

Disabling the rings in `status-mobile` behind a flag is very low effort, low risk. Such is the case because the ring is always rendered based on a single design system component. But **I wouldn't disable the rings now**, unless users face more serious bugs, then the effort to disable would be easily justifiable (at least in `status-mobile`). @churik this means we don't have to do anything now in `status-mobile`, just deprecate the feature and descope fixes related to the ring.

#### Next steps

Unless the importance of impersonation is highlighted based on user research and user feedback, I find it difficult to justify prioritizing actual development on this problem right now or even in this quarter. Although I have my own bias in favor of leveraging "mark as trusted", I believe this could end up being just patch work to a more serious problem. It would be better to solve impersonation very carefully and thoughtfully. We don't have to rush.

> Same as @jrainville, I suggest opening a separate discussion to address impersonation mitigation in Status more broadly. 

Agreed @osmaczko! 👍🏼 I'm genuinely satisfied with the outcome of our discussions: rings are deprecated, and further development on impersonation features should be thoroughly researched and aligned among more CCs. If anyone wants to continue the discussion from here, I would be happy to participate :)
